op-chain-ops/genesis/config.go

```diff
@@ -18,50 +19,260 @@ import (
+	L2ChainID                        *big.Int               `json:"l2ChainID"`
+	L2BlockTime                      uint                   `json:"l2BlockTime"`
+	MaxSequencerDrift                uint                   `json:"maxSequencerDrift"`
+	SequencerWindowSize              uint                   `json:"sequencerWindowSize"`
+	ChannelTimeout                   uint                   `json:"channelTimeout"`
+	P2PSequencerAddress              common.Address         `json:"p2pSequencerAddress"`
+	OptimismL2FeeRecipient           common.Address         `json:"optimismL2FeeRecipient"`
+	BatchInboxAddress                common.Address         `json:"batchInboxAddress"`
+	BatchSenderAddress               common.Address         `json:"batchSenderAddress"`
+	L2OutputOracleSubmissionInterval uint                   `json:"l2OutputOracleSubmissionInterval"`
+	L2OutputOracleStartingTimestamp  int                    `json:"l2OutputOracleStartingTimestamp"`
+	L2OutputOracleProposer           common.Address         `json:"l2OutputOracleProposer"`
+	L2OutputOracleOwner              common.Address         `json:"l2OutputOracleOwner"`
+	L1BlockTime                      uint64                 `json:"l1BlockTime"`
+	CliqueSignerAddress              common.Address         `json:"cliqueSignerAddress"`
+	OptimismBaseFeeRecipient         common.Address         `json:"optimismBaseFeeRecipient"`
+	OptimismL1FeeRecipient           common.Address         `json:"optimismL1FeeRecipient"`
+	GasPriceOracleOwner              common.Address         `json:"gasPriceOracleOwner"`
+	GasPriceOracleOverhead           uint                   `json:"gasPriceOracleOverhead"`
+	GasPriceOracleScalar             uint                   `json:"gasPriceOracleScalar"`
+	GasPriceOracleDecimals           uint                   `json:"gasPriceOracleDecimals"`
+	L2CrossDomainMessengerOwner      common.Address         `json:"l2CrossDomainMessengerOwner"`
+	L2GenesisBlockNonce              uint64                 `json:"l2GenesisBlockNonce"`
+	L2GenesisBlockExtraData          hexutil.Bytes          `json:"l2GenesisBlockExtraData"`
+	L2GenesisBlockGasLimit           uint64                 `json:"l2GenesisBlockGasLimit"`
+	L2GenesisBlockDifficulty         *big.Int               `json:"l2GenesisBlockDifficulty"`
+	L2GenesisBlockMixHash            common.Hash            `json:"l2GenesisBlockMixHash"`
+	L2GenesisBlockCoinbase           common.Address         `json:"l2GenesisBlockCoinbase"`
+	L2GenesisBlockNumber             uint64                 `json:"l2GenesisBlockNumber"`
+	L2GenesisBlockGasUsed            uint64                 `json:"l2GenesisBlockGasUsed"`
+	L2GenesisBlockParentHash         common.Hash            `json:"l2GenesisBlockParentHash"`
+	L2GenesisBlockBaseFeePerGas      *big.Int               `json:"l2GenesisBlockBaseFeePerGas"`
+	L1GenesisBlockTimestamp          uint64                 `json:"l1GenesisBlockTimestamp"`
```

Comment:
I'm ok-ish with this as-is, but it maybe less verbose if we use `hexutil.Uint64` and `hexutil.Big` as field types.

And if we are using `deployConfigJSON`, then the fields in this type don't need json annotations. Either one struct + typed fields, or the custom function for encoding.
Maybe we just copy over the Hive definition of the deploy config to this package. And then import that into Hive. So we only need 1 version of the deploy config Go type.